environments: ISC-inspired corrections and updates (#129)
spack/spack-tutorial: tutorial_environments.rst

   :emphasize-lines: 1

> Now let's **activate** our environment. You can use the ``spack env activate``

Reviewer: ```suggestion
Now let's **activate** our environment. You can use ``spack env activate``
```

command:

.. literalinclude:: outputs/environments/env-activate-1.out
   :language: console
   :emphasize-lines: 1

> We also automatically set up the ``spacktivate`` alias for short.

Reviewer: ```suggestion
You can also use the ``spacktivate`` alias for short.
```

> If you want to leave this environment, you can use ``spack env deactivate``.
> We also automatically set up the ``despacktivate`` alias for short.

Reviewer: ```suggestion
If you want to leave this environment, you can use ``spack env deactivate``
or the ``despacktivate`` alias for short.
```

> Now we can check the base Spack instance environment:

Reviewer: ```suggestion
After deactivating, we can see everything installed in this Spack instance:
```

I think we should not use the term 'environment' here.

     specs: [tcl, trilinos, hdf5, gmp]

.. note::

>    This setting only defines the **default** ``mpi`` provider.
>    You can override the provider in package specs. This topic
>    is covered in the :ref:`configuration tutorial <configs-tutorial>`.

Reviewer: ```suggestion
   This setting only defines the **default** ``mpi`` provider.
   You can still override the provider on the command line, e.g.,
   with`spack install hdf5 ^openmpi`. 
   
   We introduce this here to show you how environment configuration
   can affect concretization. Configuration options are covered in much
   more detail in the :ref:`configuration tutorial <configs-tutorial>`.
```

Let's see the effects of this change on our package using ``spack spec``:

.. literalinclude:: outputs/environments/spec-2.out
   :language: console
   :emphasize-lines: 1

Notice ``mpich`` is now the ``mpi`` dependency for our concretized

> spec. We also see all of its associated concrete dependencies.

Reviewer: ```suggestion
spec. We also see all of its concrete dependencies.
```

   :emphasize-lines: 1

We can demonstrate use of these environment settings by building a
really simple MPI program.

Let's create a program called ``mpi-hello.c`` that contains the following

> statements:

Reviewer: ```suggestion
code:
```

You can see that ``myproject`` is active because it is highlighted
in green.

We can also see from the listing that the environment directory
contains both of the environment files: ``spack.yaml`` and ``spack.lock``.
This is because ``spack.lock`` was generated when we concretized

> our environment.

Reviewer: ```suggestion
the environment.
```

Notice ``spack add`` added the package to our active environment and
it appears in the configuration file's spec list.

.. note::

>    This command does not install the package in your environment.

Reviewer: ```suggestion
   `spack add` does not install the package in your environment.
```
